### DISCUSSION TITLE
🎁 Share your Feedback! - Copilot Workspace for Pull Requests

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#142278] [Pull Requests]
![unnamed](https://github.com/user-attachments/assets/72606524-289d-4090-9dd8-29d75f250a0e)

Code reviews and suggestions from colleagues, integrators, and AI agents like Copilot code review and Copilot autofix increase your code’s quality, but at times they can get overwhelming. You can now use Copilot Workspace directly in the context of your pull request to quickly refine, test, and incorporate code review feedback and suggestions from teammates and AI agents. Ship faster without compromising quality.

To get access, sign up for the waitlist [here](https://gh.io/copilot-code-review-waitlist). This will also give you access to [Copilot code review](https://gh.io/copilot-code-review-public-preview-changelog).

🗳️ Want to share your feedback? Please take a moment to fill out our survey [here](https://survey3.medallia.com/?code-reviews) to help us incorporate your suggestions! As our thanks, **you’ll be entered in a draw for a $250 gift card**. Additionally, feel free to use this discussion post for quick questions, feedback, or discussion.

 - Want to know more about how Copilot Workspace came to be? Check out this Universe 2024 🛰️  from the team: [How GitHub Next took Copilot Workspace from concept to code](https://github.com/orgs/community/discussions/142971)
 -  🆕  New to GitHub and Copilot previews? Our community continues to host feature previews, visit our past discussions such as the [OpenAI o1 on GitHub (Limited Public Preview) Feedback Thread](https://github.com/orgs/community/discussions/139125) and our previous[ GitHub Copilot Workspace: Your Seamless Developer Environment discussion](https://github.com/orgs/community/discussions/139125). 
 - You may have noticed we changed our language regarding features that are in the testing and validation phases. Learn more about our terminology change in @spaltrowitz's [discussion post](https://github.com/orgs/community/discussions/141918).

**Comment by njzjz** [comment]
When I click the "update the PR" button, it doesn't push to my PR branch, but push to my main branch...

**Reply by timrogers** [reply]
@njzjz 👋🏻 Can you share a screenshot of that button so we can take a look?

**Reply by salmannizam** [reply]
@njzjz please share the screenshot.

**Reply by kwon-encored** [reply]
@Akash1134

I opted in and received access to the pre-release version of GitHub Copilot, but I don’t see 'Copilot' as an option for one of the reviewers on my Pull Request.

This is for my own private project (not related to any organization).

How do I enable the Copilot reviewer?
![image](https://github.com/user-attachments/assets/dcefee04-6b6c-453b-ba82-936a8bd08bb1)
![image](https://github.com/user-attachments/assets/6823dd39-935e-4de0-9e26-19792ef23b8f)

**Reply by SEm551** [reply]
http://bc.vc/CbRjYcP

**Comment by Starlookvideo** [comment]
Tips
I will try to describe the problem with all the details I have; these screenshots will help you to explain the problem.

You need to determine whether the problem is only on your website or is present on the site as well.

You need to reproduce the problem to understand the cause. So we need your help

Personality Quiz added with:
Multiple questions
Delete, fix, update

Improved user experience:
Clear progress indicators

Complete Personality Quiz
Delete, fix, update
Enter account details
Is there anything specific you would like to tweak about the layout or flow

Delete, fix, update I would like to tweak
I am attaching a screenshot for you, please review it and let me know the problem
Can you please show me the user interface functions image?

**Comment by salmo0** [comment]
sir .the git cloning not runing plzz . solve my problem

**Reply by BhathiyaPrasad** [reply]
add ssh key of your local machine to your GitHub account

**Comment by Raugo30** [comment]
my data was compromised somehow

**Reply by cmuto09** [reply]
Hey @Raugo30- can you share a bit more about what happened?

**Reply by ThiagoAndradeF** [reply]
Dude, here's a tip: set aside an old cell phone just to use 2FA and always carry it with you.

**Comment by 42674267Qqs2** [comment]
)))👍)

**Comment by norwd** [comment]
Please keep GenAI stuff opt-in only.

**Comment by NotAShelf** [comment]
I really don't want to see AI *anywhere* unless I *explicitly* ask for it. The new Copilot features are quite intrusive, and I do not enjoy looking at 3 Copilot buttons at any given time.

**Reply by RafaDEV99** [reply]
The new Copilot is great, but sometimes you want to control its use. Great point!

**Comment by jrasanen** [comment]
It's insanely annoying when the co-pilot widget on the front page of GitHub.com steals the focus when you access the page.

The `/` shortcut to highlight the search is broken because of this.

**Comment by 333fred** [comment]
I have absolutely no idea how the workspace editor managed this, but it somehow managed turn just clicking the apply button on [this one-line suggestion](https://github.com/dotnet/roslyn/pull/76347#discussion_r1877403908) into [this commit with 14,000 additions and 30,000 deletions](https://github.com/dotnet/roslyn/commit/5c115083e394b7b067f6ec4c5577255241a7ce61).

**Comment by tupui** [comment]
Go away with your AI! You don't respect us maintainers and steal our code without attribution and respecting our licensing terms 😡

**Comment by darikprescott** [comment]
Yeah well when my system freezes.  And my stuff is gone thru.  And no one can send a dang email or text I'm left confused on who is trying to get into my system.   I was thrown in without explicit knowledge.  But being pressured by big corporations and threatened in ways also having access to systems and things that's definitely WAY TOO MUCH power.      I didn't take any code.  I simply tried to protect it as I signed it with a pgp key and the one that was signed with me was from 2010 and didn't reply to my email.     I was trying to learn and take it slow but there was other entities trying to to make it rushed for profit and now a lot of companies are trying to give me the code when I don't want it.  I just wanted to learn and grow.  Not get handed every single tool and told good luck.   For a company to try to force itself to contract with me by simply viewing the terms of service.  These companies for far too long pick and choose and lie.  I do not intend to use any nor give it back until I have communication.   Can contact me at my email.  ***@***.***  I have sufficient evidence as to why I was impaired in my functions to attribute, and I have plenty of reasons to believe that I was apart of some confusion between these firms.   All I did was point out this SSH and the potential for abuse on Microsoft and next thing I know im turned into a dependa bit in GitHub  and having a Enterprise. When it was maybe my first week learning this stuff. .  I will and I intend to seek possible legal counsel on these matters. Not for any development or contributors but for the practices and compliance procedures as I did not break any laws and I was sure to read all terms of service.   Thank you and I do apologize for having my system freezes and also not trusting someone in my computer that I don't know trying to take code that I thought I was protecting.    Feel free to get mad or upset.  I am too.  But in the end ya gotta eat.     I took no code on purpose. I simply unplugged due to the extreme amounts of attempts to Phish, brute, and use those same tools.   Blame your sponsors, and organization's. They're the ones that gave it to me so that I can't sue.   Theyre needs to be a fix for this and I intend to make it right.  Can text me at 9404457457.  I have my name on my account.  I'm not hard to find.   Simple communication is what failed this education model.     If anyone wants to request for evidence I would be more then willing t divulge the data. 
  


Go away with your AI! You don't respect us maintainers and steal our code without attribution and respecting our licensing terms 😡

**Comment by Dekkonot** [comment]
I do not want to see generative AI anywhere I don't explicitly ask for it. If it becomes mandatory I will seriously consider moving to a different platform.

**Reply by tupui** [reply]
I am personally seriously thinking about it too. E.g. this stuff https://radicle.xyz GitLab and the rest are just copying each other at that point and will all converge to do the same crap.

**Comment by norwd** [comment]
Hey! Why is there all this Copilot stuff showing even when I have the preview turned off?!?!

The homepage now has a big fat chat box where there should be a search bar:

![image](https://github.com/user-attachments/assets/288ddb61-0ed6-4373-a09f-284269fdc69e)

And now there's a copilot icon where the "new repo" button used to be!! ![image](https://github.com/user-attachments/assets/6d8ecbe2-673d-4d7d-a705-3c0edb6d4613)

I can't help but feel that was placed so that you click it out of muscle-memory and get sucked into it. Get this dark-pattern shit out of here. I opted-out, I don't want any of these "features".

**Comment by paulshryock** [comment]
I do not want copilot or any other AI.

**Comment by remixer-dec** [comment]
I hate React as a framework, I have 0 react projects and this thing shows "REACT" suggestions on my home page even when the "feature" is disabled

UBLOCK RULE:
```
##.copilotPreview__suggestions > li :has(span:has-text("react"))
```

**Comment by WilliamDAssafMSFT** [comment]
Is not correctly reflecting changes. It's adding ^M characters to existing documents

**Reply by WilliamDAssafMSFT** [reply]
I suspect the codespace is not handling line endings correctly.

**Reply by MihaUreche** [reply]
Were you able to solve this issue?

**Reply by WilliamDAssafMSFT** [reply]
No, I had to re-do work in an external tool outside of GitHub.

**Comment by alvvaro** [comment]
I believe the setting to disable this does not work.

This is my settings page:
![Screenshot 2025-01-14 at 18 19 21](https://github.com/user-attachments/assets/8430afd2-3ea4-45f7-92fa-cfe5b0172898)

This is what shows on the PR page:
![Screenshot 2025-01-14 at 18 19 37](https://github.com/user-attachments/assets/2fa04db6-d322-41d8-ad20-26953e03548a)
Plus a two copilot buttons on every. single. line.

@Akash1134, why was this enabled automatically? This should be opt-in only

**Comment by titaniumtraveler** [comment]
I do not want Copilot would like to disable it.
This includes the UI element in the AppHeader:
![github-copilot-ui-1](https://github.com/user-attachments/assets/3e657404-ff15-4a11-84a4-cffef262b61d)

And the UI in the dashboard:
![github-copilot-ui-2](https://github.com/user-attachments/assets/f518a13a-1d1d-42aa-9945-8140cec8204d)

They are definitely a good feature for the people who use Copilot, but for people who *don't* it leads to visual clutter and reduces usability.

From what I know there are settings to disable these elements under `Settings > Code, planning, and automation > Copilot`, but those don't seem to be available when not having enabled Copilot.

Possible Solutions:

The best solution in my opinion is to keep the UI elements enabled by default, but *only* when having enabled Copilot.

An alternative solution would be to make the settings to disable the Copilot UI-elements available even if not having enabled Copilot.

**Comment by MashaMSFT** [comment]
i don't like the new codespaces editor and i dont' want to be forced to use it, please make it easy to go to back to the classic experience, the instructions provided by copilot to go back to the classic experience are inaccurate

**Comment by ThiagoAndradeF** [comment]
It's incredibly frustrating when the co-pilot widget on GitHub's homepage takes focus as soon as you access the page. This also breaks the / shortcut for highlighting the search bar.

**Comment by RDIL** [comment]
Please **kindly stop putting anything AI** in the product without me explicitly saying "yes, I want to use this". I hate this experiment with every bone in my body.

**Comment by figadore** [comment]
How do I disable this and go back to the normal in-browser editor? When I try to edit a file in the PR, it's doing things I don't understand and don't like. Suddenly there are git conflicts that I have to resolve, but I don't know where they're coming from. I just want to make a one-character change to an existing file in an existing PR.

**Comment by ccdredzik** [comment]
How can I disable it?

**Reply by ThiagoAndradeF** [reply]
Dude, when I find out I'll tell you, haha, it's going to be a great day

**Reply by titaniumtraveler** [reply]
I can recommend `UBlockOrigin` with HTML filters.
(See <https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#html-filters>)

I use these:
```
github.com##^div.copilotPreview__container
github.com##^div.AppHeader-CopilotChat
```

I *used* to use normal cosmetic filters, but because they work after the first
render, they would show you the copilot UI elements for a split second, which
can be annoying.

The HTML filters instead filter *before* the browser renders the first time,
which is awesome!

It is sad that github doesn't give us any reasonable option to disable all that, but that doesn't mean there aren't ways to disable them anyways.

**Comment by KrysztofN** [comment]
Sure, it's nice to have quick fixes for simple issues, but now I'm drowning in a mix of human reviews, AI suggestions, and automated checks all fighting for attention in the same space. What happened to thoughtful code review discussions?

**Comment by ccdredzik** [comment]
When I am adding an empty new line at the end of a file it makes an empty commit (without the new line). Garbage.

**Comment by fdhoiwu3o8** [comment]
The editor is so jacked up for me. Sometimes when I apply a simple change I'll get an error screen, only to refresh and see the changes did actually commit. 

When making changes and also using suggestions from colleagues (that appear in the sidebar, but nowhere else in my pull request), it adds a lot of empty lines, some duplicate lines, and even deletes unrelated lines.

It's jacked up, please make this stuff opt-in. When people are using this comment section to recommend uBlock HTML filtering (funnily enough by linking the source code on GH itself) specifically to disable it, then you know it's getting bad.

**Comment by karunpoudel-chr** [comment]
One feature (may be unintentional) that I like in traditional github web edit, which is missing in Copilot Workspace, is being able to push a commit without making any changes (type a character and delete it to enable commit button).
Use case: Sometimes, randomly, when we push a commit in a PR, our azure pipeline doesn't start. In this case, making an empty commit does the trick.
(There is "Run pipeline" button in azure pipeline to trigger manually from there but it doesn't work properly for PR because `system.pullRequest.pullRequestNumber` would be null in such case.)

**Comment by deganmee** [comment]
I dislike the default commit message, `Updates from editor`, too generic. at least include filename so message indicates impacted files.

**Comment by Forgind** [comment]
It would be nice to have an option to break long lines onto multiple lines (visibly, not actually changing the code) for when I edit a .md file.

**Comment by kwon-encored** [comment]
@Akash1134

I opted in and received access to the pre-release version of GitHub Copilot, but I don’t see 'Copilot' as an option for one of the reviewers on my Pull Request.

This is for my own private project (not related to any organization).

How do I enable the Copilot reviewer?
![image](https://github.com/user-attachments/assets/dcefee04-6b6c-453b-ba82-936a8bd08bb1)
![image](https://github.com/user-attachments/assets/6823dd39-935e-4de0-9e26-19792ef23b8f)

**Reply by natepmay** [reply]
Same here.

**Comment by koppor** [comment]
Jump to definition does not work. As a reviewer, I need to dig into the larger context. I do not want to open my Desktop IDE, but quickly use the web browser for "quicker" things. And "jump to definition" would be a quick thing. The "web editor" (github.dev) could be opened when ctrl+click on a method is done.

**Comment by TaxBusby** [comment]
Really enjoying this feature as I edit files from the web UI a lot and this gives me a more convenient experience vs editing a single file, or opening a full `github.dev` IDE.

I would like to see this feature more accessible from other contexts. For example, how can I go from a page where I'm viewing a file on a branch, to editing that file in a copilot workspace to draft a new PR?

Right now I start a draft PR by editing one file, and then make the rest of my changes in the workspace UI from the PR diff.

**Comment by Jackuczys** [comment]
The editing experience in new workspace seems nice but I only learnt about this because I misclicked - I wouldn't have even tried clicking on it normally because I have no interest in AI aspect of it, just in the better editor UI (while not having to use something as big as github.dev) and I auto-assume that features with copilot logo are just various ways to prompt AI to do things with no other kind of use cases.

**Comment by martincostello** [comment]
I've noticed that sometimes when I edit files in the new UX, `^M` characters get added to new lines and mess up the formatting. The non-preview experience for editing files doesn't do this.

**Comment by adamhoffstein** [comment]
Awful. How do i switch back?

**Comment by mcclure** [comment]
There is now a "Code with Copilot Agent Mode" button on issue pages for me. It appears to have the purpose of creating a pull request or a branch. I'm not sure if the correct place to report this is the Pull Requests or Issues forum, but I am extremely bothered by this and I want to remove it from both my own interface and any issue page attached to one of my projects. If the "Code with Copilot Agent Mode" could create a pull request (or create a branch which could later become a pull request) then this would necessarily violate the contributor terms of my projects (I do not accept code written by Copilot). For this reason, other users should not be given the copilot options either when browsing a project I control.

This is what I am referring to:

![image](https://github.com/user-attachments/assets/4e1b11a1-f2f8-445f-91c2-e5476f1d49d8)

**Comment by yoksel** [comment]
Markdown hotkeys does not work in editor

**Comment by jnewland** [comment]
I clicked 'Edit file' on a PR and was surprised to see a status message in the bottom right saying "Starting codespace..." on the page I was taken to. I didn't ask for that at all and it seems incredibly wasteful? I was literally fixing a typo in a text document.

**Comment by chuggafan** [comment]
Can I never see AI stuff on my webpage again?
Forcing AI down users throats is exactly the indication that billions have been spent on a technology that no one uses and I have to constantly make moves in Microsoft products (including github) to disable Copilot/Cortana/ChatGPT/Claude/Whatever it is this time constantly. It's a rather annoying constant in my life at this point.

At the very least, make it *opt-in* instead of *opt-out*.

Thank you.

**Comment by darikprescott** [comment]
How the hell do we stop this harassment by these corporations that derive money from people who can't afford lawyers or more.      I'm sick and tired of the harm and the bullsh.      It was easier a year ago when you could crack a program and use it for free.   Now they want to steal the stuff from your device and let ai run wild with it 

Yahoo Mail: Search, Organize, Conquer 
 
  On Thu, Apr 17, 2025 at 5:01 PM, ***@***.***> wrote:   


Can I never see AI stuff on my webpage again?
Forcing AI down users throats is exactly the indication that billions have been spent on a technology that no one uses and I have to constantly make moves in Microsoft products (including github) to disable Copilot/Cortana/ChatGPT/Claude/Whatever it is this time constantly. It's a rather annoying constant in my life at this point.

At the very least, make it opt-in instead of opt-out.

Thank you.

**Comment by paulshryock** [comment]
Do not want.

**Comment by SEm551** [comment]
http://bc.vc/CbRjYcP

**Comment by ghostinhershell** [comment]
Hi all 👋🏾 

Thank you for the detailed report regarding "Code with Copilot Agent Mode" appearing in the UI even though it's disabled and hidden in the settings. We’ve reviewed your notes in this discussion and have already passed this along to the relevant engineering team, where it’s being tracked internally for investigation.

We appreciate your thoroughness and the time you’ve taken to help surface this. Improving reliability and user experience is a priority, and reports like yours are incredibly helpful.

Thanks again for your contribution to the platform!

**Comment by jc-usda** [comment]
I selected copilot as the reviewer of a PR. Copilot did a nice job of summarizing the PR - even included a table of the files changed in its "Pull Request Overview" comment. Then, Copilot found a place where i had missed a space after a comma (I didn't lint!), and it suggested a change. The comma in question was in arguments listed in a function call, like:

```
# blah blah
# roxygen stuff
my_function <- function(arg1, arg2, 
                                        arg3,arg4) {
# actual function 
# blah
}
```

And it said, hey, you should have a space before arg4, and it had a new file up, but only showed that line. 
However, in the new file, what it saved with its cowritten commit was:
```
# blah blah
# roxygen stuff
                                        arg3, arg4) {
# actual function 
# blah
}
```
and now the critical `my_function <- function(arg1, arg2, ` was gone.
